OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -629,6 +630,165 @@ public static DreamValue NativeProc_clamp(DreamObject instance, DreamObject usr,
             }
             return new DreamValue(0);
         }
+        
+        [DreamProc("gradient")]
+        [DreamProcParameter("A", Type = DreamValueType.DreamObject)]
+        [DreamProcParameter("index", Type = DreamValueType.Float)]
+        public static DreamValue NativeProc_gradient(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
+            if (arguments.ArgumentCount <= 1) {
+                throw new IndexOutOfRangeException("Not enough arguments");
+            }
+
+            /// We dont want keyword arguments screwing with this
+            DreamValue indx = arguments.GetArgument(arguments.OrderedArguments.Count - 1, "index");
+
+            int colorspace = 0;
+            bool loop = false;
+
+            if (!indx.TryGetValueAsFloat(out float index)) {
+                throw new FormatException("Failed to parse index as float");
+            }
+
+            List<DreamValue> GradientList;
+
+            if (arguments.GetArgument(0, "A").TryGetValueAsDreamList(out DreamList gradlist)) {
+                GradientList = gradlist.GetValues();
+
+                if (gradlist.ContainsValue(new DreamValue("loop"))) { loop = true; }
+
+                DreamValue dictspace = gradlist.GetValue(new("space"));
+                if (dictspace != DreamValue.Null && dictspace.TryGetValueAsInteger(out int dictlookup)) { colorspace = dictlookup; }
+
+            } else {
+                GradientList = arguments.GetAllArguments();
+            }
+
+            if (arguments.OrderedArguments.Contains(new("loop"))) { loop = true; }
+
+            if (arguments.NamedArguments.TryGetValue("space", out DreamValue namedlookup) && namedlookup.TryGetValueAsInteger(out int space)) { colorspace = space; }
```

Comment:
This should be moved below the `GradientList = arguments.GetAllArguments();` above because this should only happen if you don't provide a color gradient list.

This can also be replaced with
```cs
loop = GradientList.Contains(new("loop"));
if (arguments.NamedArguments.TryGetValue("space", out DreamValue namedLookup))
    namedLookup.TryGetValueAsInteger(out colorspace);
```